lib/philips.js

```diff
@@ -123,15 +141,26 @@ function decodeGradientColors(input, opts) {
         const yLow = parseInt(input.slice(0, 2), 16) << 8;
         input = input.slice(2);
 
-        const x = xHigh | xLow;
-        const y = yHigh | yLow;
 
+        const x = Math.round((xHigh | xLow) / 65535 * 10000) / 10000;
+        const y = Math.round((yHigh | yLow) / 65535 * 10000) / 10000;
+
+        if (mode === COLOR_MODE_COLOR_XY) {
+            return {
+                color_mode: 'xy',
+                x, y,
+                brightness, on,
+            };
+        }
+
+        // Effect mode
+        const effect = input.slice(0, 4);
+        const name = knownEffects[effect] || `unknown_${effect}`;
         return {
-            color_mode: 'xy',
-            x: Math.round(x / 65535 * 10000) / 10000,
-            y: Math.round(y / 65535 * 10000) / 10000,
-            brightness,
-            on,
+            color_mode: 'effect',
```

Comment:
For compatibility reasons we have to stick to the Home Assistant color modes: https://www.home-assistant.io/integrations/light.mqtt/#json-schema . I propose to leave this `xy`

Reply:
Cool! I'll change it! :-)